### ISSUE TITLE
CCScale9Sprite::setColor(...) is broken.

### ISSUE THREAD

**Issue by edbartley** [cocos2d/cocos2d-x#2242]
Adding these lines to the end of the HelloWorld::init() method with this image in the resources directories
![drop_frame](https://f.cloud.github.com/assets/1588880/287262/fcb51a38-9252-11e2-89e2-93810aaeb511.png)
should result in a large green square with rounded corners in the center of the screen and a small red circle in the center of the the green square. Unfortunately, the CCScale9Sprite will not change color to green it will remain white.

```
CCScale9Sprite *scaleSprite = CCScale9Sprite::create("drop_frame.png");
scaleSprite->setPreferredSize(CCSizeMake(250, 250));
scaleSprite->setColor(ccGREEN);
scaleSprite->setPosition(ccp(origin.x + visibleSize.width/2, origin.y + visibleSize.height/2));
this->addChild(scaleSprite);    

CCSprite *sp = CCSprite::create("drop_frame.png");
sp->setColor(ccRED);
sp->setPosition(ccp(origin.x + visibleSize.width/2, origin.y + visibleSize.height/2));
this->addChild(sp);
```

**Comment by edbartley** [comment]
You'll also need to add the Extensions directory to the project, obviously.

**Comment by Clarinexus** [comment]
This is also happening when loaded through a .ccbi file, since it uses the same setColor function. CCSprites are unaffected.

**Comment by tiantian20007** [comment]
CCScale9Sprite::setOpacity is also not working;

**Comment by dumganhar** [comment]
Thank you. I'll check it.

**Comment by dumganhar** [comment]
Issue http://www.cocos2d-x.org/issues/1876 was created.

**Comment by dumganhar** [comment]
Fixed at https://github.com/cocos2d/cocos2d-x/pull/2373. Thanks for your feedback.

**Comment by smilingpoplar** [comment]
@dumganhar Please revert back setColor()/setOpacity() for `CCProgressTimer` also.

**Comment by Clarinexus** [comment]
This is still not fixed as of the May 1st release of Cocos2dx 2.1.3.
